### DISCUSSION TITLE
Problem installing 2.0.248 - Code 19 Windows cannot start this hardware device

### DISCUSSION THREAD

**Discussion by pupphelper** [nefarius/DsHidMini#135] [Q&A]
Hello all, Having issues with the newest driver, version 1.4 installs and works okay.  However when installing the latest driver, both .infs.   Rebooting, and doing clean installs as well as following the other troubleshooting steps, like resetting the controller etc.  I was unable to get the driver to start.

"Windows cannot start this  hardware device because its configuration information (in the registry) is incomplete or damaged. (Code 19)

Additional notes.  I started with 2.0.248 without having any previous drivers versions of the driver installed, so it should have been clean.  I went to 1.4 only after ver 2.0 failed.
I have cleaned the driver store and done full uninstalls of all devices and the driver.  I did not have scptoolkit, fireshock or the sixaxxis dll previous.  I did however have DS4Windows installed, but I uninstalled the HideHid before trying this driver.   Unfortunately I keep getting the same error.   Do you have any suggestions. Are there any registry entries you would like to see from my system?

![image](https://user-images.githubusercontent.com/10311986/131253654-a5495140-7bcf-4c24-a476-af973ad4ba63.png)

**Comment by nefarius** [comment]
Try using [this tool](https://www.uwe-sieber.de/files/DeviceCleanup.zip) to remove everything resembling the controller, then start over. Make extra sure you read the instructions carefully and only use versions documented to be compatible with each other. Good luck!

**Comment by DocMAX** [comment]
i have the same problem. tool didn't help.
edit: sorry missed the igfilter.inf. works now thanks!

**Comment by AetherSurfer85** [comment]
I just encountered the same problem on my Windows 10 (64 bit) PC.  I used the linked "DeviceCleanup.exe" tool to remove the controller, which appeared in device manager as "DS3 Compatible HID Device."  I cleaned the driver store to remove dshidmini and igfilter, using DriverStoreExplorer v0.11.92.  I then updated BthPS3 to the latest version 2.6.174 (there was an older version on my system from a long time ago, probably leftover from before I reinstalled Windows for a new motherboard.)  I reinstalled the dshidmini and igfilter drivers from the .inf files from version 2.2.282.0.  Upon plugging in my PS3 controller via USB, I am still getting the code 19 error.  I tried another controller just to be safe, and I got the same error.  I also tried pushing the reset button on the controller, which did not work either.  I rebooted many times during the uninstall/reinstall process.

It might be worth noting that I have DS4Windows, ViGEmBus, HidHide, and FakerInput installed on my system, in case this affects things.  When I first plugged in the PS3 controller, before installing dshidmini, Windows recognized it as a PS3 controller, but it was constantly disconnecting and reconnecting.  It would briefly appear in DS4Windows as well.  Several years ago, I had SCPToolkit installed, but I uninstalled it following the instructions on your site quite some time ago.  This was before I switched over to DS4Windows.  I can't find any traces of SCPToolkit on my system anymore.

Can you give me any advice on how to fix this problem?  I really want to get analog button support working.  Thank you.

**EDIT:**

I managed to solve this problem now!  It was not an issue with dshidmini, but rather a problem on my system.  For anyone else encountering this error, here is what fixed it for me:

1.  I ran sfc /scannow from an elevated command prompt, and it reported that there were errors.  It corrected them.
2.  Plugging the controller in again did NOT work right away!
3.  I uninstalled the devices and drivers, following Nefarius's instructions from his site.
4.  I reinstalled the drivers from the dshidmini.inf and igfilter.inf files.
5.  I plugged the PS3 controller back in, and at first a windows error message popped up saying that device could not start.  I unplugged the controller and tried my other PS3 controller, and it was recognized and showed up in DSHMC.exe correctly!  I tried the first controller again, and it too was detected.  I'm not sure if the second controller mattered, or if I could have just plugged in the first controller a second time.

Note that no reboot was required for this.

I hope that this helps anyone else who has this problem!  Many thanks to Nefarius for providing this wonderful tool, and for his detailed instructions on the install/uninstall process.

**Reply by Kanuan** [reply]
Nice to know that sfc /scannow may be used to try solving the issue

**Reply by nefarius** [reply]
Wait, are you telling me sfc can actually fix things?! Dang! I thought it was just an empty executable all these years 🤣
